cmd/optimize/events.go

```diff
@@ -460,6 +482,32 @@ func listRecommendations(flags *recommendationsCmdFlags) func(*cobra.Command, []
 			_, next_ok = data_set.Links["next"]
 		}
 
+		// extract recommendation optimizer ID + num to query and append blocker data from optimization_started events
+		for i := range recommendationRows {
+			blockerRows, err := getOptimizationBlockerData(recommendationRows[i], flags)
```

Comment:
While this approach technically works, I don't think its really scalable. The optimization_started event query takes 5-30 seconds for each row and with 10s to 100s of recommendations the runtime can become quite long.

Instead, I would recommend fetching all relevant `optimization_started` events in a single query. Forming the query for this should be straightforward as you can simply pass `tempVals` (same passed tothe recommendation query) into your optimization_started template as is (extra properties of the struct not used by your query template will simply be ignored).

As for processing the resulting query, I would recommend writing a modified version of extractEventsData; instead of returning a slice, this version returns a map with a string key in the form of "{optimizer_id}-{optimizer_num}" with the value being a map of the optimization_started event attributes.

Once you've retrieved the optimization_started events, you can then iterate the recommendation rows and associate them to the relevant started_event by constructing the relevant key from the recommendation row data.